libs/SalesforceNetwork/SalesforceNetwork/SalesforceNetwork/Classes/Network/Action/CSFSalesforceAction.m

```diff
@@ -209,6 +209,16 @@ - (void)observeValueForKeyPath:(NSString *)keyPath ofObject:(id)object change:(N
     if (context == kObservingKey) {
         [self willChangeValueForKey:@"ready"];
         [self didChangeValueForKey:@"ready"];
+        if ([keyPath isEqualToString:@"communityId"]) {
+            self.enqueuedNetwork.defaultConnectCommunityId = self.enqueuedNetwork.account.communityId;
+        } else if (self.enqueuedNetwork.account.credentials.accessToken
```

Comment:
This needs to be subject to the value of the `requiresAuthentication` property as well.